src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs

```diff
@@ -407,44 +407,49 @@ public async Task UpdatePullRequestAsync(string pullRequestUri, PullRequest pull
-        /// <param name="parameters">Settings for merge</param>
-        /// <returns></returns>
-        public async Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters)
+        public async Task<IList<Commit>> GetPullRequestCommitsAsync(string pullRequestUrl)
         {
             (string accountName, string projectName, string repoName, int id) = ParsePullRequestUri(pullRequestUrl);
-
             VssConnection connection = CreateVssConnection(accountName);
             GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
-
-            string commitToMerge = parameters.CommitToMerge;
-
-            // If the commit to merge is empty, look it up first.
-            if (string.IsNullOrEmpty(commitToMerge))
+            var pullRequest = await client.GetPullRequestAsync(repoName, id, includeCommits: true);
+            IList<Commit> commits = new List<Commit>(pullRequest.Commits.Length);
+            foreach (var commit in pullRequest.Commits)
             {
-                var prInfo = await client.GetPullRequestAsync(repoName, id);
-                commitToMerge = prInfo.LastMergeSourceCommit.CommitId;
+                commits.Add(new Commit(commit.Author.Name == "DotNet-Bot" ? "dotnet-maestro[bot]" : commit.Author.Name,
+                    commit.CommitId,
+                    commit.Comment));
             }
+            connection.Dispose();
+            return commits;
+        }
+
+        public async Task MergeDependencyPullRequestAsync(string pullRequestUrl, MergePullRequestParameters parameters, string mergeCommit)
+        {
+            (string accountName, string projectName, string repoName, int id) = ParsePullRequestUri(pullRequestUrl);
+
+            VssConnection connection = CreateVssConnection(accountName);
+            GitHttpClient client = await connection.GetClientAsync<GitHttpClient>();
+            var pullRequest = await client.GetPullRequestAsync(repoName, id , includeCommits:true);
 
             await client.UpdatePullRequestAsync(
                 new GitPullRequest
                 {
                     Status = PullRequestStatus.Completed,
                     CompletionOptions = new GitPullRequestCompletionOptions
                     {
+                        MergeCommitMessage = mergeCommit,
                         BypassPolicy = true,
                         BypassReason = "All required checks were successful",
                         SquashMerge = parameters.SquashMerge,
                         DeleteSourceBranch = parameters.DeleteSourceBranch
                     },
-                    LastMergeSourceCommit = new GitCommitRef { CommitId = commitToMerge }
+                    LastMergeSourceCommit = new GitCommitRef { CommitId = pullRequest.LastMergeSourceCommit.CommitId, Comment = mergeCommit }
                 },
                 projectName,
                 repoName,
                 id);
+            connection.Dispose();
```

Comment:
Yep, the using up there would mean this could be removed.  They key bit about the using is that it's really:
```
try
{
 x = new myIDisposableObject();
}
finally
{
  if (x != null)
  {
    x.Dispose();
  }
}
```

So no matter whether an exception is thrown or not, the object will be disposed.